src/components/scenes/GuiPluginListScene.tsx

```diff
@@ -149,23 +156,56 @@ class GuiPluginList extends React.PureComponent<Props, State> {
 
   async updatePluginsNetwork(diskPlugins: BuySellPlugins | undefined) {
     const { coreDisklet } = this.props
-    let networkPlugins: BuySellPlugins
+    const diskPluginsJson = JSON.stringify(diskPlugins)
+
+    // Convert to and from JSON so we don't overwrite the default object
+    let currentPluginsJson = diskPlugins != null ? diskPluginsJson : buySellPluginsJson
+    const currentPlugins = asBuySellPlugins(JSON.parse(currentPluginsJson))
+
+    // Grab plugin settings that patch the json
     try {
-      const response = await fetchInfo(`v1/buySellPlugins/${config.appId ?? 'edge'}`)
+      const response = await fetchInfo(`v1/buySellPluginsPatch/${config.appId ?? 'edge'}`)
       const reply = await response.json()
-      networkPlugins = asBuySellPlugins(reply)
-      if (JSON.stringify(networkPlugins) !== JSON.stringify(diskPlugins)) {
-        await coreDisklet
-          .setText(PLUGIN_LIST_FILE, JSON.stringify(networkPlugins))
-          .then(() => (diskPlugins = networkPlugins))
-          .catch(e => console.error(e.message))
-        this.setPluginState(networkPlugins)
+      const networkPluginsPatch = asBuySellPlugins(reply) ?? {}
+      const directions: Array<'buy' | 'sell'> = ['buy', 'sell']
+      for (const direction of directions) {
+        const patches = networkPluginsPatch[direction]
+        if (patches == null) {
+          // Assign the defaults
+          currentPlugins[direction] = buySellPlugins[direction]
+          continue
+        }
+        const currentDirection = currentPlugins[direction] ?? []
+        if (currentPlugins[direction] == null) {
+          currentPlugins[direction] = currentDirection
+        }
```

Comment:
This section doesn't really make sense or I'm missing something

        // If currentPlugins[direction] == null, currentDirection is []
        const currentDirection = currentPlugins[direction] ?? []

        // At the same time, if currentPlugins[direcion] == null, currentPlugins[direction] = []
        if (currentPlugins[direction] == null) {
          currentPlugins[direction] = currentDirection
        }